compiler/src/dmd/func.d

```diff
@@ -1677,9 +1689,16 @@ extern (C++) class FuncDeclaration : Declaration
      */
     extern (D) final void setThrow(Loc loc, const(char)* fmt, RootObject arg0 = null)
     {
-        if (nothrowInprocess && !nothrowViolation)
+        if (nothrowInprocess)
         {
-            nothrowViolation = new AttributeViolation(loc, fmt, arg0); // action that requires GC
+            if (semanticRun < PASS.semantic3 && _scope)
+            {
+                this.semantic2(_scope);
+                this.semantic3(_scope);
+            }
+
+            if (nothrowInprocess && !nothrowViolation)
```

Comment:
```suggestion
            if (!nothrowViolation)
```
Redundant condition

Reply:
It's not redundant. After semantic pass it sets this to false if the semantic passes happens to resolve it.